Sming/Components/OtaUpgrade/OtaUpgradeStream.cpp

```diff
@@ -14,18 +14,19 @@
 #ifdef OTA_ENCRYPTED
 #include <sodium/utils.h>
 #endif
+#include <FlashString/Array.hpp>
 
-// symbol provided by appcode (generated source file)
+// Keys defined by `appcode/keys.cpp` which is compiled with application
 #ifdef OTA_SIGNED
-extern "C" uint8_t OTAUpgrade_SignatureVerificationKey_P[] PROGMEM;
+DECLARE_FSTR_ARRAY(OTAUpgrade_SignatureVerificationKey, uint8_t);
```

Comment:
After some debugging, it seems `DECLARE_FSTR_ARRAY` is not compatible with `IMPORT_FSTR_ARRAY`, because the former declares a reference
```
extern const FSTR::Array<ElementType> /* --> */ & /* <-- */ name;
```
and the latter a "by-value" symbol:
```
extern "C" const FSTR::Array<ElementType> name;
```
This should probably be added to the documentation somehow.
Or how do you feel about extending `IMPORT_FSTR_ARRAY` like this:
```cpp
#define IMPORT_FSTR_ARRAY(name, ElementType, file) \
	IMPORT_FSTR_DATA(name##_imported, file) \
	extern "C" const FSTR::Array<ElementType> name##_imported; \
	const FSTR::Array<ElementType>& name = name##_imported
```

Reply:
I did cover this issue with [IMPORT_FSTR](https://sming.readthedocs.io/en/latest/_inc/Sming/Components/FlashString/utility.html#c.IMPORT_FSTR_DATA) but didn't properly document the newer `IMPORT_FSTR_ARRAY` version.

However, your suggestion of defining an additional reference makes a great deal of sense, simplifying the documentation and making it easier to use. Thank you! I'll update the library.

Reply:
OK, I've updated the FlashString library. It's a pity we still need the C symbol for linkage...